programs/mmm/src/instructions/mpl_core_asset/mpl_core_wrap.rs

```diff
@@ -37,3 +51,66 @@ impl Deref for IndexableAsset {
         &self.0
     }
 }
+
+pub fn deserialize_collection_asset(
+    collection_account: &UncheckedAccount,
+) -> Result<Option<IndexableAsset>, Error> {
+    if collection_account.key == &Pubkey::default() {
+        return Ok(None);
+    }
+    let data = &collection_account.try_borrow_data()?;
+    if let Ok(collection_asset) =
+        mpl_core::IndexableAsset::fetch(mpl_core::types::Key::CollectionV1, data)
+            .map(IndexableAsset)
+    {
+        return Ok(Some(collection_asset));
+    }
+    Err(MMMErrorCode::InvalidAssetCollection.into())
+}
+
+pub fn get_creators_from_royalties(royalties: &Royalties) -> Vec<Creator> {
+    royalties
+        .creators
+        .iter()
+        .map(|creator| Creator {
+            address: creator.address,
+            verified: true,
+            share: creator.percentage,
+        })
+        .collect()
+}
+
+pub fn get_royalties_from_plugin(
+    asset: &IndexableAsset,
+    collection: Option<&IndexableAsset>,
+) -> Option<Royalties> {
+    let asset_royalty_plugin = asset.plugins.get(&PluginType::Royalties);
```

Comment:
I believe this is actually the only place that we read data from the collection. Let's make sure that the collection is actually the collection that the asset belongs to.

Reply:
we actually did check these in https://github.com/magicoss/mmm/blob/37d0f8378b1a0f51e9a99ea4424120e638fe8286/programs/mmm/src/instructions/mpl_core_asset/mpl_core_fulfill_buy.rs#L242

I'm hesitate add in this function because it make code a bit more complicated that has to handle the error when call this function